Multi-country targeting campaign - Pre-populate the audience field

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

This PR implemented the "**Pre-populating the Audience field**" part of #1248 and based on #1299.

- Pre-populate the **Audience** field with the countries that the merchant previously selected as part of their Merchant Center setup in the campaign setup and campaign creation page

### Screenshots:

https://user-images.githubusercontent.com/17420811/157216052-10be698e-4c27-448d-874d-16a1f2739cc2.mp4

### Detailed test instructions:

1. Go to campaign setup directly with path `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-ads`
   - Even if the setup is completed, it doesn't need to disconnect first, it can be entered directly with the path. 🙈 
1. The **Audience** field should be pre-populated with the countries that were previously selected in the onboarding setup or free listings edit page.
1. Go to the campaign creation page.
1. Same as 2. The **Audience** field should be pre-populated.

### Changelog entry

eason9487 committed:
Pre-populate the audience field on campaign creation

eason9487 committed:
Pre-populate the audience field on Campaign Setup

eason9487 commented on js/src/setup-ads/setup-ads-form.js:
```diff
@@ -12,26 +12,25 @@ import { getNewPath } from '@woocommerce/navigation';
 import validateForm from '.~/utils/paid-ads/validateForm';
 import { recordLaunchPaidCampaignClickEvent } from '.~/utils/recordEvent';
 
-// when amount is null or undefined in an onChange callback,
-// it will cause runtime error with the Form component.
```
📝  This comment was removed because it's no longer applicable after #670.

puntope reviewed (commented):
❓  Unsure about if this is intentional or not. 

When you select "All Countries" in the Campaigns. Then when you add the new Paid Campaign it doesn't show all the Countries.  


https://user-images.githubusercontent.com/5908855/158173903-61d24e36-499e-44f1-a3af-eb914e15e984.mov

eason9487 committed:
Fix that pre-populating countries are incorrect when selecting all countries as target audience.

Address https://github.com/woocommerce/google-listings-and-ads/pull/1300#pullrequestreview-908720163

eason9487 commented:
@puntope, thanks for catching this! 🙏 

> When you select "All Countries" in the Campaigns. Then when you add the new Paid Campaign it doesn't show all the Countries.

Yes, that was a bug. Fixed in https://github.com/woocommerce/google-listings-and-ads/pull/1300/commits/740b811272f688590013f87599bf776b10294ae9. Could you help with the new round of code review?

puntope reviewed (approved):
✅  LGTM 

Thanks for the updates. tested again and it works as expected.
